service-base/src/main/java/org/eclipse/hono/service/registration/AbstractRegistrationService.java

```diff
@@ -335,4 +422,38 @@ protected JsonArray getSupportedGatewaysForDevice(final String tenantId, final S
         return viaObj instanceof JsonArray ? (JsonArray) viaObj : new JsonArray(Collections.emptyList());
     }
 
+    /**
+     * Gets the list of gateways that may act on behalf of the given device.
+     * <p>
+     *
+     * This default implementation gets the list of gateways from the value of the
+     * {@link RegistrationConstants#FIELD_VIA} property in the device's registration information.
+     * If a gatewayId is already known (e.g. through an assert request) this device is also included in the return value.
+     * If no acting gateway is known and the {@link RegistrationConstants#FIELD_VIA} is empty the groups in the
+     * {@link RegistrationConstants#FIELD_VIA_GROUPS} property in the device's registration information are resolved and
+     * returned.
```

Comment:
This sentence is also not reflected in the current implementation of this PR. My proposal is to stick to the current implementation and to always resolve the groupMembers of the groups mentioned in the `memberOf´ property of the device. This mitigates possible side effects where the content of the `via` value in the response of the `assert` endpoint of the Device Registration API may change drastically depending on the presence of a value in the `via` property of the device in question. 
However, always resolving the group introduces additional overhead. In many situations this overhead may even be unnecessary if the caller only needs to identify one possible gateway and is not interested in all available devices. At the same time, a part of this overhead could be mitigated by the usage of caching mechanisms.
WDYT?

Reply:
> My proposal is to stick to the current implementation and to always resolve the groupMembers of the groups mentioned in the `memberOf` property of the device.

I agree. The result of the `assertRegistration` request is cached anyway in the protocol adapters, so the additional overhead is probably not very noticable anyway, IMHO.